src/App.tsx

```diff
@@ -1,26 +1,77 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from './app/store';
+import { useEffect, useMemo, useState } from 'react';
+import { getTodos } from './api';
+import { setTodosList } from './features/todos';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const dispatch = useDispatch();
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const todoList = useSelector((state: RootState) => state.todos);
+  const filter = useSelector((state: RootState) => state.filter);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoading(true);
+      try {
+        const todosGetter = await getTodos();
+
+        dispatch(setTodosList(todosGetter));
+      } catch {
+        alert('May be some problem with todos loading. Try again.');
```

Comment:
Avoid using 'alert' for error handling. It's better to have a dedicated error handling mechanism, like a dedicated error state in your Redux store. You can then display this error in a user-friendly manner in your UI.